task/prepare.go

```diff
@@ -0,0 +1,80 @@
+package task
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/golang/glog"
+	livepeerAPI "github.com/livepeer/go-api-client"
+	"github.com/livepeer/go-livepeer/drivers"
+	"github.com/livepeer/stream-tester/model"
+	"github.com/livepeer/stream-tester/segmenter"
+)
+
+func RecordStream(ctx context.Context, lapi *livepeerAPI.Client, playbackID string, osSess drivers.OSSession) (string, error) {
+	glog.Infof("made it into record function")
+	fullPath := videoFileName(playbackID)
+	glog.Infof("full path %s", fullPath)
+	fir, err := osSess.ReadData(ctx, fullPath)
+	glog.Infof("inputOS %b", osSess.IsExternal())
+	if err != nil {
+		return "", fmt.Errorf("error reading data from source OS url=%s err=%w", fullPath, err)
+	}
+	sourceFile, err := readFile(fir)
+	fir.Body.Close()
+	if err != nil {
+		return "", err
+	}
+	defer sourceFile.Close()
+	var sourceFileSize int64
+	if fir.Size != nil {
+		sourceFileSize = *fir.Size
+	}
+
+	streamName := fmt.Sprintf("vod_recording_%s", time.Now().Format("2006-01-02T15:04:05Z07:00"))
+	stream, err := lapi.CreateStreamEx(streamName, true, nil)
```

Comment:
Might want to use `CreateStreamEx2` to set custom profiles instead. I say we process the info from the `asset` here, especially resolution, to create a specific list of profiles that we want the asset in. Could be as simple as sending the regular 240p, 480p, 720p, 1080p (the transcoders will handle keeping the aspect ration in the response), but removing any profiles with a resolution and/or bitrate higher than the current asset. WDYT?

Reply:
Yup that makes sense as well, actually for testing I am only importing a 480p file since my computer cannot transcode any higher - so it is wrong to be outputting 1080p, 720p etc

Reply:
Can we defer this to a follow-up PR as well?

Reply:
I think it's a pretty weird experience to send in a 480p file and get 720p and 1080p back, especially since they will not have a better quality. As we chatted offline I sent a suggestion of what this logic could look like, I think it wouldn't be that complex.

If we can't get it to work for the deploy now it's ok to start without it tho